packages/dev/core/src/Materials/shadowDepthWrapper.ts

```diff
@@ -240,22 +242,29 @@ export class ShadowDepthWrapper {
             fragmentCode = origEffect.fragmentSourceCodeBeforeMigration;
 
         if (!this.doNotInjectCode) {
-            // vertex code
+            const shaderLanguage = (this._baseMaterial as ShaderMaterial).options?.shaderLanguage ?? ShaderLanguage.GLSL;

```

Comment:
We should not need to know that the material is a `ShaderMaterial`...

The easiest way would be to add a `shaderLanguage` getter on `Effect` and simply use `origEffect.shaderLanguage` to get the shader language.

Reply:
`origEffect.shaderLanguage` is applied now.